File: hw7/GTUMap.java
import javafx.util.Pair;

/**
 * Make a map class
 * @author ferhat
 * @param <K> first type for Map
 * @param <V> Second type for Map
 */
public class GTUMap<K,V> extends GTUSet<Pair<K,V>> {
    
     /**
     *
     * @return value for corresponding key if exists 
     */
    V at(K k){
        
        Pair<K,V> temp =null;
        GTUIterator<Pair<K,V>> p =begin();
        while(p.hasNext()){  // looking for that element
            temp =p.next();
            if(temp.getKey() == k){        
                break;
            }
        }
        
        return temp.getValue();
    }
    
}

File: hw6/GTUMap.cpp
#include "GTUMap.h"

namespace GTUFerhat{

    template<class K,class V>
    GTUMap<K,V>::GTUMap():GTUSet<pair<K,V> >(){
    
    }
    
    template<class K,class V>
    V& GTUMap<K,V>::operator[](const K& k){
        
        typename GTUMap<K,V>::GTUIterator p =this->begin(); 
        for(;p !=this->end();++p){ // looking for that element
        
            if(p->first == k){
                return p->second;
            }
        }
        this->insert(pair<K,V>(k,V())); // if k value does not exist then insert new one and return
        return p->second;
    }

}


